File: 20. Valid Parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        pair = {
            ')': '(',
            ']': '[',
            '}': '{'
        }
        left_bracket = set( pair.values() )

        for char in s:
            if char in left_bracket:
                stack.append(char)
            elif not stack or (stack[-1] != pair[char]):
                return False
            else:
                stack.pop()

        return len(stack) == 0

File: 20. Valid Parentheses/solution.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Boolean
     */
    function isValid($s) {
        $arr = [
            '(' => ')',
            '{' => '}',
            '[' => ']',
        ];
        $stack = [];
        for ($i=0;$i<strlen($s);$i++) {
            if (array_key_exists($s[$i], $arr)) {
                $stack[] = $s[$i];
            } else if (false === ($idx = array_search($s[$i], $arr))) {
                return false;
            } else if ($idx != array_pop($stack)) {
                return false;
            }
        }

        return empty($stack) ? true : false;
    }
}